spec/controllers/councillor_contributions_controller_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+
+describe CouncillorContributionsController do
+    let(:authority){create(:authority)}
+    let(:councillor_contribution){create(:councillor_contribution, authority: authority)}
+    let(:suggested_councillors){create(:suggested_councillor, councillor_contribution: councillor_contribution)}
```

Comment:
> when I use expect(response.body).to include it only returns name,start_date,end_date,exective,council,council_website,id,email,image,party,source,ward,phone_mobile, and not the name and email of the suggested councillor. It is probably my way of setting up dataset is no right. any suggestions?

I suspect that the reason for that is that you're using the [`let` method](https://relishapp.com/rspec/rspec-core/docs/helper-methods/let-and-let) here. `let` lazy loads the data, waiting for you to call it:

> Note that let is lazy-evaluated: it is not evaluated until the first time
> the method it defines is invoked. You can use let! to force the method's
> invocation before each example.

—[rspec docs](https://relishapp.com/rspec/rspec-core/docs/helper-methods/let-and-let)

Notice that while the `authority` and `councillor_contribution` variables get called in that setup, `suggested_councillors` doesn't get called anywhere. That means it never gets loaded.

Because this is just one test, you'd be fine to inline all of this into the `it` block as assign what needs to be assigned as local variables. That's cool cause this setup isn't being used by any other tests so there's no shared code.

Another solution would be to use `let!` rather than `let` for the SuggestedCouncillor method.

The difference between `let` and `let!` has tripped me up plenty of times :)